kernel/memory/src/paging/mapper.rs

```diff
@@ -537,18 +621,45 @@ impl MappedPages {
             return Ok(());
         }
 
-        for page in self.pages.range().clone() {
-            let p1 = active_table_mapper.p4_mut()
-                .next_table_mut(page.p4_index())
-                .and_then(|p3| p3.next_table_mut(page.p3_index()))
-                .and_then(|p2| p2.next_table_mut(page.p2_index()))
-                .ok_or("mapping code does not support huge pages")?;
-            
-            p1[page.p1_index()].set_flags(new_flags);
+        match self.pages.page_size() {
+            PAGE_4KB_SIZE => {
+                for page in self.pages.range().clone() {
+                    let p1 = active_table_mapper.p4_mut()
+                        .next_table_mut(page.p4_index())
+                        .and_then(|p3| p3.next_table_mut(page.p3_index()))
+                        .and_then(|p2| p2.next_table_mut(page.p2_index()))
+                        .ok_or("remap: error getting p1 entry for 4kb page")?;
+
+                    p1[page.p1_index()].set_flags(new_flags);
+
+                    tlb_flush_virt_addr(page.start_address());
+                }
+            }
+            PAGE_2MB_SIZE => {
+                for page in self.pages.range_2mb().clone().into_iter().step_by(512) {
```

Comment:
the same comment from above applies to all of the blocks where you iterate over the huge page ranges -- there shouldn't be any need to use `step_by` or a 2MiB-specific function like `range_2mb()`

Reply:
Yeah I'm not a fan of using `into_iter` and `step_by` here.

But the reason I went with this is because the `PageRange` is ultimately just a large `RangeInclusive` of 4KiB pages. If you had 2 huge pages, the range that's looped over would cover all the pages in between the 2 huge pages that we actually want to map.

As an alternative to this, would it work to change `Step` implementation for `Page` and `Frame` in order to adapt to the different sizes?